### PR TITLE
Support dell rpl products

### PR DESCRIPTION
**Pull request by gongjun-song**

### CODE REVIEW

File: tools/topology/topology1/CMakeLists.txt [hunk 1]
```diff
@@ -131,6 +131,7 @@ set(TPLGS
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l0-rt1316-l13-rt714-l2\;-DPLATFORM=adl\;-DUAJ_LINK=0\;-DAMP_1_LINK=1\;-DAMP_2_LINK=3\;-DEXT_AMP_REF\;-DMIC_LINK=2"
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l2-rt1316-l01-rt714-l3\;-DPLATFORM=adl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DMIC_LINK=3"
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l2-rt1316-l01\;-DPLATFORM=adl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DNO_LOCAL_MIC"
+	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-rpl-rt711-l2-rt1316-l01-rt714-l3\;-DPLATFORM=rpl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DMIC_LINK=3"
```
Comment (plbossart) [inline tools/topology/topology1/CMakeLists.txt:134]: You also need to update the Soundwire topologies in kernel_dependent/v5.19/CMakelist.txt

Unfortunately we have two sets of definitions to add dynamic pipelines and deep-buffer conditionally.

**Review by plbossart — CHANGES_REQUESTED** [review]
looks good but you need to take care of the updated topologies as well, see below.

**Review by RanderWang — APPROVED** [review]
LGTM

File: tools/topology/topology1/CMakeLists.txt (hunk 1 above)
Comment (gongjun-song) [reply tools/topology/topology1/CMakeLists.txt:134]: @plbossart Have updated, please help review, thanks

**Review by lgirdwood — APPROVED** [review]

File: tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt [hunk 2]
```diff
@@ -34,6 +34,7 @@ set(TPLGS_UP
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l0-rt1316-l13-rt714-l2\;-DHEADSET_DEEP_BUFFER\;-DDYNAMIC=1\;-DPLATFORM=adl\;-DUAJ_LINK=0\;-DAMP_1_LINK=1\;-DAMP_2_LINK=3\;-DEXT_AMP_REF\;-DMIC_LINK=2"
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l2-rt1316-l01-rt714-l3\;-DHEADSET_DEEP_BUFFER\;-DDYNAMIC=1\;-DPLATFORM=adl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DMIC_LINK=3"
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l2-rt1316-l01\;-DHEADSET_DEEP_BUFFER\;-DDYNAMIC=1\;-DPLATFORM=adl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DNO_LOCAL_MIC"
+	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-rpl-rt711-l2-rt1316-l01-rt714-l3\;-DHEADSET_DEEP_BUFFER\;-DDYNAMIC=1\;-DPLATFORM=rpl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DMIC_LINK=3"
```
Comment (plbossart) [inline tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt:37]: this is good but you didn't update the previous patch, it needs 2 lines in two different files as well.

**Review by plbossart — CHANGES_REQUESTED** [review]

File: tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt (hunk 2 above)
Comment (gongjun-song) [reply tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt:37]: @plbossart I'm sorry, I didn't quite understand what you meant.

One of Dell's platforms has no headset. I only need to add the one platform with the headset to the two CMakeLists.txt respectively. Is there something wrong with my understanding?

Comment (gongjun-song) [reply tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt:37]: @plbossart 
Do you mean we need to add product configuration in 5.19 CMakeLists.txt even if the product doesn't have headset?
If yes, I think we are missing config for some ADL products in 5.19 CMakeLists.txt.

Comment (plbossart) [reply tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt:37]: @gongjun-song there are two additions in this 'kernel-dependent' stuff

HEADSET_DEEP_BUFFER is only relevant indeed for platforms with a jack.
DYNAMIC=1 is required for all platforms.

That means that if we are missing ADL products, we need to add them as a correction. We should have exactly the same number of entries for SoundWire devices in tools/topology/topology1/CMakeLists.txt and tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt

Can you please double-check if indeed we are missing some devices? Thanks!

Comment (gongjun-song) [reply tools/topology/topology1/kernel_dependent/v5.19/CMakeLists.txt:37]: @plbossart  Thanks for the clarification.
I add a new commit to fix some missing adl topologies in v5.19/CMakeLists.txt. Please help review

**Review by plbossart — APPROVED** [review]
thanks @gongjun-song

**Comment by plbossart** [comment]
kernel changes merged in https://github.com/thesofproject/linux/pull/3978

**Pull request merged**